### DISCUSSION TITLE
BlazingMQ on Hacker News

### DISCUSSION THREAD

**Discussion by quarter-note** [bloomberg/blazingmq#69] [Announcements]
Hi everyone!

There was a good discussion about BlazingMQ on Hacker News when it was published as open source.  Hacker News thread can be found [here](https://news.ycombinator.com/item?id=36896034).

It was great to see a lively and meaningful discussion on the product!

Thank you!

**Comment by s3rius** [comment]
Hi! When do you want to publish python library? We want to create BlazingMQ integration for @taskiq-python.

**Reply by hallfox** [reply]
Hi @s3rius, thanks for the interest in BlazingMQ! We're expecting to have things ready to release the Python SDK in the next two months or so, however this is a rough timeline. We do have a very good idea of what it will look like and what features it will support though, so if you have any requirements for integrating it into taskiq I'd be happy to discuss. Namely we won't have asyncio support out of the box, as it will be based on bindings to the native client library.

**Reply by pniedzielski** [reply]
@s3rius We've just open-sourced the Python SDK!  You can find it here: https://github.com/bloomberg/blazingmq-sdk-python and once our build kicks off it should be in PyPI as well.

**Reply by s3rius** [reply]
Since Python SDK doesn't have an asyncio support. I have a question. Do you have a protocol documentation? 

https://github.com/bloomberg/blazingmq/blob/main/src/groups/bmq/bmqp/bmqp_protocol.h

I only found this header. Is there any documentation page about it? We can try to implement a native library in Rust with python bindings. Like we did it here: https://github.com/Intreecom/scyllapy.

**Reply by pniedzielski** [reply]
We have a draft document that doubles as both protocol documentation and some ideas for how to structure a robust client library [in this PR](https://github.com/bloomberg/blazingmq/pull/95).  I'd still say `bmqp_protocol` is a good component to have as a reference, because it describes the wire protocol in great depth, which you would need if you are implementing a native library in Rust.

**Reply by s3rius** [reply]
This seems like a comprehensive guide about the protocol. We will take a look about implementing it. Thank you.
